protocol/version.go

```diff
@@ -40,23 +40,21 @@ func IsSupportedVersion(supported []VersionNumber, v VersionNumber) bool {
 }
 
 // HighestSupportedVersion finds the highest version number that is both present in other and in SupportedVersions
-// the versions in other do not need to be ordered
 // it returns true and the version number, if there is one, otherwise false
-func HighestSupportedVersion(other []VersionNumber) (bool, VersionNumber) {
-	var otherSupported []VersionNumber
-	for _, ver := range other {
+func HighestSupportedVersion(ours, theirs []VersionNumber) (bool, VersionNumber) {
+	var theirsSupported []VersionNumber
+	for _, ver := range theirs {
 		if ver != VersionUnsupported {
-			otherSupported = append(otherSupported, ver)
+			theirsSupported = append(theirsSupported, ver)
```

Comment:
I think we don't need this at all. Before this PR, the `PublicHeader` would parse a Version Negotiation Packet such that unsupported versions (i.e. versions for which we didn't have a `protocol.VersionNumber` defined) were replaced by `VersionUnsupported`.
I changed that in one of the commits in the PR, since "supported" / "unsupported" now has two different meanings, if the user specifies the acceptable versions in the `quic.Config`. Thus, we don't need this check here anymore.